lib/spack/docs/packaging_guide.rst

```diff
@@ -1441,6 +1441,32 @@ The ``when`` clause follows the same syntax and accepts the same
 values as the ``when`` argument of
 :py:func:`spack.directives.depends_on`
 
+^^^^^^^^^^^^^^^
+Sticky Variants
+^^^^^^^^^^^^^^^
+
+The variant directive can be marked as ``sticky`` by setting to ``True`` the
+corresponding argument:
+
+.. code-block:: python
+
+   variant('bar', default=False, sticky=True)
+
+A ``sticky`` variant differs from a regular one in that it is always set
+to either:
+
+#. An explicit value appearing in a spec literal or
+#. Its default value
+
+The concretizer thus is not free to pick an arbitrary value to work
```

Comment:
```suggestion
The concretizer thus is not free to pick an alternate value to work
```

I don't think we want to suggest the concretizer chooses "arbitrary" variant values

Reply:
Fixed